src/utils.tsx

```diff
@@ -0,0 +1,10 @@
+import {AuthorizationStatus} from './constants';
+import { Result } from './types';
+
+export const authorization = (auth: AuthorizationStatus, trueResult: Result, falseResult: Result) => {
+  if (auth === AuthorizationStatus.Auth) {
+    return trueResult;
+  }
+
+  return falseResult;
```

Comment:
```suggestion
  return auth === AuthorizationStatus.Auth ? trueResult : falseResult;
```

Но я бы оставил эту утилиту просто возвращающую `true/false`, а вот конкретные `result` уже использовал бы снаружи